pwiz_tools/Skyline/TestPerf/DiaUmpireTutorialTest.cs

```diff
@@ -0,0 +1,922 @@
+            Assert.IsTrue(searchSucceeded.Value);
+
+            var addIrtDlg = ShowDialog<AddIrtPeptidesDlg>(() => importPeptideSearchDlg.ClickNextButton(), 30 * 60000);//peptidesPerProteinDlg.OkDialog());
+            RunUI(() =>
+            {
+                // Check values shown in the tutorial
+                Assert.AreEqual(1, addIrtDlg.RunsConvertedCount);
+                var row = addIrtDlg.GetRow(0);
+                Assert.AreEqual(11, row.Cells[1].Value);
+
+                var regressionLine = new RegressionLine(_analysisValues.IrtSlope, _analysisValues.IrtIntercept);
+                if (!IsRecordMode)
+                {
+                    Assert.AreEqual(_analysisValues.LibraryPeptideCount, addIrtDlg.PeptidesCount);
+                    Assert.AreEqual(regressionLine.DisplayEquation, row.Cells[2].Value);
+                }
+                else
+                {
+                    _analysisValues.LibraryPeptideCount = addIrtDlg.PeptidesCount;
+                    Console.WriteLine($@"LibraryPeptideCount = {addIrtDlg.PeptidesCount},");
+                    Console.WriteLine(ParseIrtProperties(row.Cells[2].Value.ToString()));
+                }
+
+                Assert.AreEqual(1.0, double.Parse(row.Cells[3].Value.ToString()));
+                Assert.AreEqual(Resources.AddIrtPeptidesDlg_AddIrtPeptidesDlg_Success, row.Cells[4].Value);
+            });
+            PauseForScreenShot("Add iRT peptides form", 15);
+
+            var irtGraph = ShowDialog<GraphRegression>(() => addIrtDlg.ShowRegression(0));
+            PauseForScreenShot("iRT regression graph", 15);
+
+            OkDialog(irtGraph, irtGraph.CloseDialog);
+            var recalibrateMessage = ShowDialog<MultiButtonMsgDlg>(addIrtDlg.OkDialog);
+            RunUI(() => Assert.AreEqual(TextUtil.LineSeparate(Resources.LibraryGridViewDriver_AddToLibrary_Do_you_want_to_recalibrate_the_iRT_standard_values_relative_to_the_peptides_being_added_,
+                Resources.LibraryGridViewDriver_AddToLibrary_This_can_improve_retention_time_alignment_under_stable_chromatographic_conditions_), recalibrateMessage.Message));
+
+            if (!_instrumentValues.HasAmbiguousMatches)
+            {
+                OkDialog(recalibrateMessage, recalibrateMessage.ClickNo);
+            }
+            else
+            {
+                var ambiguousDlg = ShowDialog<MessageDlg>(recalibrateMessage.ClickNo);
+                RunUI(() => AssertEx.Contains(ambiguousDlg.Message,
+                    Resources.BiblioSpecLiteBuilder_AmbiguousMatches_The_library_built_successfully__Spectra_matching_the_following_peptides_had_multiple_ambiguous_peptide_matches_and_were_excluded_));
+                OkDialog(ambiguousDlg, ambiguousDlg.OkDialog);
+            }
+            OkDialog(addIrtDlg, addIrtDlg.OkDialog);
+
+            var peptidesPerProteinDlg = ShowDialog<PeptidesPerProteinDlg>(() => {}, 600000); //importPeptideSearchDlg.ClickNextButton(), 600000);
```

Comment:
You should use "WaitForOpenForm" here instead of "ShowDialog".

"ShowDialog" will fail if the dialog happens to already be showing.